docker/dockerfiles/artifacts/artifact.Dockerfile

```diff
@@ -51,4 +51,11 @@ COPY --from=fe-builder /build/starrocks/output /release/fe_artifacts
 COPY --from=be-builder /build/starrocks/output /release/be_artifacts
 COPY --from=broker-builder /build/starrocks/fs_brokers/apache_hdfs_broker/output /release/broker_artifacts
 
+# download the latest dd-java-agent
+ADD 'https://dtdg.co/latest-java-tracer' /release/fe_artifacts/fe/datadog/dd-java-agent.jar
+
+# Get ddprof for BE profiling
+RUN wget https://github.com/DataDog/ddprof/releases/download/v0.15.3/ddprof-0.15.3-amd64-linux.tar.xz -O ddprof-amd64-linux.tar.xz && \
```

Comment:
would be better to make the versions as argument and default to latest.
Then when version upgrade, we don't need another PR to change the docker file

Reply:
There isn't a URL point to the latest version similar as `https://dtdg.co/latest-java-tracer`, create an issue to DD team https://github.com/DataDog/ddprof/issues/368

Reply:
be sure the url will work for both arm and amd64. this dockerfile will be used for both arm and x86_64 cpu arch.

Reply:
it can be found on their github release page: https://github.com/DataDog/dd-trace-java/releases
You just need to replace the version number in the middle of the url:
E.g. 
https://github.com/DataDog/dd-trace-java/releases/download/v1.28.0/dd-java-agent.jar
https://github.com/DataDog/dd-trace-java/releases/download/v1.27.0/dd-java-agent.jar

Reply:
> be sure the url will work for both arm and amd64. this dockerfile will be used for both arm and x86_64 cpu arch.

On yes. they have both AMD and ARM versions:
https://github.com/DataDog/ddprof/releases

Reply:
> it can be downloaded from their github release page: https://github.com/DataDog/dd-trace-java/releases You just need to replace the version number in the middle of the url: E.g. https://github.com/DataDog/dd-trace-java/releases/download/v1.28.0/dd-java-agent.jar https://github.com/DataDog/dd-trace-java/releases/download/v1.27.0/dd-java-agent.jar

For Java, `https://dtdg.co/latest-java-tracer` would be the best URL, it always points to the latest version. I don't think it's necessary to pin on a version, then either keep updating it or provide it during docker image build.

Reply:
> be sure the url will work for both arm and amd64. this dockerfile will be used for both arm and x86_64 cpu arch.

Good point, let me fix it.